.github/workflows/danger.yml

```diff
@@ -32,10 +32,20 @@ jobs:
         # frozen-lockfile will make the build fail if the lockfile is not there
         run: yarn install --frozen-lockfile
 
-      - name: Run Consistency Checks
+      - name: Validate Labels
+        run: |
+          yarn run danger ci \
+            --dangerfile Automattic/peril-settings/org/pr/label.ts \
+            --no-publish-check \
+            --id pr_labels
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Consistency Checks
         run: |
           yarn run danger ci \
             --dangerfile Automattic/peril-settings/org/pr/ios-macos.ts \
+            --no-publish-check \
```

Comment:
with this option | without this option
--- | ---
| <img width="970" alt="Screen Shot 2020-02-19 at 16 43 58" src="https://user-images.githubusercontent.com/1218433/74805699-05467900-5338-11ea-89d9-54e523c85c7a.png">

Reply:
This is interesting! 

My only thought what happens in this scenario:

1) User pushes to the repo, and those checks run.
2) User adds labels, then those checks run.

Does the second set of tests hide the results of the first? Or does it just update the list for the second set? IMHO we'd wouldn't want to hide the old ones when the new ones run.

Reply:
I have a theory on what would happen. But let me run a test and see 😄

Reply:
All right... So, if I understood your question properly, the answer is: no, the n-th check doesn't hide the result of the previous n+1 checks.

To test it, I pushed a new commit...

<img width="785" alt="Screen Shot 2020-02-20 at 21 53 28" src="https://user-images.githubusercontent.com/1218433/74927313-76b32400-542b-11ea-9977-febdd6d9d6da.png">

...then changed the labels while the build was still running...

<img width="805" alt="Screen Shot 2020-02-20 at 21 54 28" src="https://user-images.githubusercontent.com/1218433/74927383-99453d00-542b-11ea-9ec5-83c7e5a58dba.png">

...and the result was one single message and 2 checks reports 

<img width="939" alt="Screen Shot 2020-02-20 at 21 51 57" src="https://user-images.githubusercontent.com/1218433/74927470-c85bae80-542b-11ea-9d7d-a6d4a05a1a62.png">


I think this is a good behavior. Having a link to all the checks that run on the current `HEAD` of the branch gives us full visibility.

Reply:
To be clear, what `--no-publish-check` does is not publishing an _extra_ check result with the Danger run. 

![](https://user-images.githubusercontent.com/1218433/74805699-05467900-5338-11ea-89d9-54e523c85c7a.png)

What we see here, without the `--no-publish-check` is:

- One check result for the whole workflow
- One check result for the Danger run for the consistency checks as part of the workflow
- One check result for the Danger run for the PR labels as part of the workflow

Once all of [the Danger rules from the peril-settings repo](https://github.com/Automattic/peril-settings/blob/5d22258ca46c3649f2cb863cfac2d16f12c4cba0/peril-settings.json#L10-L22) are in place, we'd potentially have 1 check result for the whole workflow plus 5 checks for each rule. I think that might become a bit noisy. I feel they'd be noisy because if any of the Danger runs doesn't fully pass, Danger will post a comment anyways.

Reply:
That sounds sensible to me – if the entire thing is rolled up into one check with an exit status code for pass/fail, that sounds great. 

And if it fails, it'll post a comment, which is exactly what we want.

So I think this is great and can be opened as a PR on the main repo. Thanks!

Reply:
Will do, just waiting for the PR this is based off to be merged too. Thanks for following up on this 😄 🙏 .